### PR TITLE
PXC-3551: enforce_storage_engine doesn't work on startup with PXC

### PR DESCRIPTION
**Pull request by dutow**
Issue: enforce_storage_engine ignores some tables, but wsrep tables
aren't on the list. Because of this, enforcing any other engine than
innodb will prevent the creation of wsrep tables on startup.

Fix: add an exception for wsrep tables.

### CODE REVIEW

File: sql/sql_table.cc [hunk 1]
```diff
@@ -18893,14 +18893,17 @@ static bool check_engine(THD *thd, const char *db_name, const char *table_name,
...
     */
     bool no_substitution = (!is_engine_substitution_allowed(thd));
 
     bool enforcement_forbidden =
         ((thd->lex->sql_command == SQLCOM_ALTER_TABLE) &&
          (create_info->used_fields & HA_CREATE_USED_ENGINE) == 0) ||
         (thd->lex->sql_command == SQLCOM_OPTIMIZE) ||
-        dd::get_dictionary()->is_dd_table_name(db_name, table_name);
+        dd::get_dictionary()->is_dd_table_name(db_name, table_name) ||
+        (strncmp(db_name, "mysql", 6) == 0 &&
```
Comment (venkatesh-prasad-v) [inline sql/sql_table.cc]: Can we make use of an already existing `is_wsrep_system_table()` function here?

Ref: https://github.com/dutow/percona-xtradb-cluster/blob/pxc3551/sql/table.h#L4096-L4103

Comment (dutow) [reply sql/sql_table.cc]: Updated.

**Review by venkatesh-prasad-v — APPROVED** [review]
Thanks for addressing the review comments, Approved!

**Review by kamil-holubicki — APPROVED** [review]
lgtm

**Pull request merged**